python/lsst/ts/scriptqueue/script_queue.py

```diff
@@ -353,34 +355,68 @@ def report_summary_state(self):
         if enabled:
             self.do_showAvailableScripts()
 
+    def put_next_visit(self, script_info):
+        """Output the ``nextVisit`` event.
+        """
+        if self.verbose:
+            print(f"put_next_visit: index={script_info.index}, "
+                  f"group_id={script_info.group_id}")
+        if script_info.metadata is None:
+            raise RuntimeError("script_info has no metadata")
+        if not script_info.group_id:
+            raise RuntimeError("script_info has no group_id")
+        metadata_dict = dict((key, value) for key, value in script_info.metadata.get_vars().items()
```

Comment:
Consider the more compact form of dict comprehension:
```python
metadata_dict = {key: value for key, value in script_info.metadata.get_vars().items()
                 if not key.startswith("private_")}
```
I think that looks clearer because `key: value` looks more dict-like than a tuple.